source/fe/fe_poly_tensor.cc

```diff
@@ -84,66 +83,82 @@ namespace internal
                              ExcInternalError());
 
                       // TODO: This is probably only going to work for those
                       // elements for which all dofs are face dofs
-                      face_sign[f * dofs_per_face + j] = -1.0;
+                      if ((mapping_type.size() > 1 ?
+                             mapping_type[cell_j] :
+                             mapping_type[0]) == mapping_raviart_thomas)
+                        face_sign[f * fe.dofs_per_face + j] = -1.0;
                     }
               }
           }
       }
 
 
 
+      template <int spacedim>
       void
       get_face_sign_change_rt(
         const dealii::Triangulation<3>::cell_iterator & /*cell*/,
-        const unsigned int /*dofs_per_face*/,
-        std::vector<double> &face_sign)
+        const FiniteElement<3, spacedim> & /*fe*/,
+        const std::vector<MappingType> & /*mapping_type*/,
+        std::vector<double> & /*face_sign*/)
       {
-        std::fill(face_sign.begin(), face_sign.end(), 1.0);
         // TODO: think about what it would take here
       }
 
+
+
+      template <int spacedim>
       void
       get_face_sign_change_nedelec(
         const dealii::Triangulation<1>::cell_iterator & /*cell*/,
-        const unsigned int /*dofs_per_face*/,
-        std::vector<double> &face_sign)
+        const FiniteElement<1, spacedim> & /*fe*/,
+        const std::vector<MappingType> & /*mapping_type*/,
+        std::vector<double> & /*face_sign*/)
       {
         // nothing to do in 1d
-        std::fill(face_sign.begin(), face_sign.end(), 1.0);
       }
 
 
 
+      template <int spacedim>
       void
       get_face_sign_change_nedelec(
         const dealii::Triangulation<2>::cell_iterator & /*cell*/,
-        const unsigned int /*dofs_per_face*/,
-        std::vector<double> &face_sign)
+        const FiniteElement<2, spacedim> & /*fe*/,
+        const std::vector<MappingType> & /*mapping_type*/,
+        std::vector<double> & /*face_sign*/)
       {
-        std::fill(face_sign.begin(), face_sign.end(), 1.0);
         // TODO: think about what it would take here
       }
 
 
+      template <int spacedim>
       void
       get_face_sign_change_nedelec(
         const dealii::Triangulation<3>::cell_iterator &cell,
-        const unsigned int /*dofs_per_face*/,
-        std::vector<double> &face_sign)
+        const FiniteElement<3, spacedim> & /*fe*/,
+        const std::vector<MappingType> &mapping_type,
+        std::vector<double> &           face_sign)
       {
         const unsigned int dim = 3;
-        std::fill(face_sign.begin(), face_sign.end(), 1.0);
         // TODO: This is probably only going to work for those elements for
         // which all dofs are face dofs
         for (unsigned int l = 0; l < GeometryInfo<dim>::lines_per_cell; ++l)
-          if (!(cell->line_orientation(l)))
+          if (!(cell->line_orientation(l)) & mapping_type[0] == mapping_nedelec)
```

Comment:
Did you mean to use `&` instead of `&&` here? I think it will produce the same result since both are `bool`s but I would prefer to use the logical operators.

Reply:
Whoops. I got it, thanks!